pkg/iterator/emitters.go

```diff
@@ -76,7 +81,112 @@ func (e *TableEmitter) SnapshotMetadataIteratorRecord(recordNumber int, metadata
 		fmt.Fprintf(e.Writer, tableRowFmt, recordNumber, metadata.VolumeCapacityBytes, bmt, bmd.ByteOffset, bmd.SizeBytes)
 	}
 
-	return true
+	return nil
+}
+
+func (e *TableEmitter) SnapshotMetadataIteratorDone(_ int) error {
+	return nil
+}
+
+// VerifierEmitter will write the changed blocks from the source device to the target device at the designated offset and
+// compare the contents of the source and target devices at the end.
+type VerifierEmitter struct {
+	// SourceDevice contains the source device file descriptor.
+	SourceDevice *os.File
+
+	// TargetDevice contains the target device file descriptor.
+	TargetDevice *os.File
 }
 
-func (e *TableEmitter) SnapshotMetadataIteratorDone(_ int) {}
+// copyBlock will copy chunks of 256 bytes at a time upto to sizeBytes from the source device to the target device.
+func (verifierEmitter *VerifierEmitter) copyBlock(bmd *api.BlockMetadata) error {
+	// Seek to the byteOffset of the source device.
+	_, err := verifierEmitter.SourceDevice.Seek(bmd.ByteOffset, io.SeekStart)
+	if err != nil {
+		return fmt.Errorf("failed to seek source device(%q) to offset(%d): %w", verifierEmitter.SourceDevice.Name(), bmd.ByteOffset, err)
```

Comment:
Just a suggestion: It would be good if you defined an error `ErrValidationFailed` and used it as follows:
```
return fmt.Errorf("%w: failed ...", ErrValidationFailed, ...)
```
Then in your UTs you could distinguish between errors from this package versus, say, the `io` package and deliberately test for the error type with something like this:
```
assert.ErrorIs(t, err, ErrValidationFailed)
```